src/data/providers/balancer-api/modules/pool-state/index.ts

```diff
@@ -0,0 +1,90 @@
+import { BalancerApiClient } from '../../client';
+import { PoolStateInput } from "../../../../../entities";
+
+export class Pools {
+  readonly poolStateQuery = `query GetPool($id: String!){
+    poolGetPool(id:$id) {
+      id
+      address
+      name
+      type
+      version
+      ... on GqlPoolWeighted {
+        tokens {
+          ... on GqlPoolTokenBase {
+            address
+             decimals
+            index
+          }
+        }
+      }
+      ... on GqlPoolStable {
+        tokens {
+          ... on GqlPoolTokenBase {
+            address
+             decimals
+            index
+          }
+        }
+      }
+      ... on GqlPoolPhantomStable {
+        tokens {
+          ... on GqlPoolTokenBase {
+            address
+             decimals
+            index
+          }
+        }
+      }
+      ... on GqlPoolGyro {
+        tokens {
+          ... on GqlPoolTokenBase {
+            address
+             decimals
+            index
+          }
+        }
+      }
+      ... on GqlPoolLiquidityBootstrapping {
+        tokens {
+          ... on GqlPoolTokenBase {
+            address
+             decimals
+            index
+          }
+        }
+      }
+      ... on GqlPoolElement {
+        tokens {
+          ... on GqlPoolTokenBase {
+            address
+             decimals
+            index
+          }
+        }
+      }
+      ... on GqlPoolLiquidityBootstrapping {
+        tokens {
+          ... on GqlPoolTokenBase {
+            address
+             decimals
+            index
+          }
+        }
+      }
+    }
+}`;
+
+  constructor(private readonly balancerApiClient: BalancerApiClient) {}
+
+  async fetchPoolState(id: string): Promise<PoolStateInput> {
```

Comment:
Thinking ahead to next version, there won't be poolId. Is it possible to filter on address instead?

Reply:
With the API we have now, it's not possible with the query "poolGetPools", but is possible with the query "poolGetPools", the problem is that "poolGetPools" returns a pool model that cannot retrieve the token index, so we have 2 options:

1 - Create a query similar to poolGetPool that accept address
2 - Adapt the query poolGetPools to retrieve index

But either way we will have to make changes in the API.

Reply:
Thanks for this info. Will create an Asana ticket to handle this.